### PR TITLE
ci: ensure conventional commit styles are used for PRs

### PR DESCRIPTION
**Pull request by ChildishGiant**
See #79. Also bumps version in package-lock

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Developer change (build systems etc).

## Checklist:

- [x] I have read the **[CONTRIBUTING document](https://github.com/materializecss/materialize/blob/master/CONTRIBUTING.md)**.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by warrenrodrigues — APPROVED** [review]

File: .husky/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_
```
Comment (DanielRuf) [inline .husky/.gitignore:1]: What does this line do?

File: .github/workflows/lint-commits.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name: Lint Commit Messages
+on: [pull_request]
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (DanielRuf) [inline .github/workflows/lint-commits.yml:10]: Do we really have to fetch all commits, tags and so on?

File: .husky/commit-msg [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
```
Comment (DanielRuf) [inline .husky/commit-msg:2]: Is this the code that is generated by husky? I've used husky in the past but this is new to me.

File: .husky/commit-msg [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+npx --no-install commitlint --edit $1 --color
```
Comment (DanielRuf) [inline .husky/commit-msg:4]: Same here, is this code generated by husky? In my opinion commitlint should be added as a devDependency. In this case we do not need npx afaik.

File: .github/workflows/lint-commits.yml (hunk 2 above)
Comment (ChildishGiant) [reply .github/workflows/lint-commits.yml:10]: The readme for the action I'm using specifies that this is needed so all commits on the PR are linted, not just the latest one. (https://github.com/wagoid/commitlint-github-action#usage)

File: .husky/commit-msg (hunk 4 above)
Comment (ChildishGiant) [reply .husky/commit-msg:4]: Like the other files you've commented on, these were generated by husky so I haven't touched them (other than adding the color arg). It seems that this is the standard way of doing things (https://github.com/typicode/husky/blob/main/.husky/commit-msg) but this is my first time using husky so I'm not certain on this.

Comment (DanielRuf) [reply .husky/commit-msg:4]: Ah ok, thx for the clarification.

**Pull request merged**